llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2572,6 +2572,46 @@ Instruction *InstCombinerImpl::foldICmpSRemConstant(ICmpInst &Cmp,
   return new ICmpInst(ICmpInst::ICMP_UGT, And, ConstantInt::get(Ty, SignMask));
 }
 
+Instruction *InstCombinerImpl::foldICmpRemConstant(ICmpInst &Cmp,
+                                                   BinaryOperator *Rem,
+                                                   const APInt &C) {
+  const ICmpInst::Predicate Pred = Cmp.getPredicate();
+  Value *X = Rem->getOperand(0);
```

Comment:
```suggestion
  if (!ICmpInst::isEquality(Pred) || !C.isZero())
    return nullptr;
  Value *X = Rem->getOperand(0);
```
It only holds for `==/!= 0`.